src/Service/ServiceExe.cs

```diff
@@ -93,15 +96,47 @@ public static string ServiceExeLogFolder
                               options.Port = port;
                               options.IncludeFailureDetailsInResponse = true;
                               options.MaxConcurrentCalls = 5;
+                              options.SslCertificate = GetCertificate();
+                              options.EnableSsl = true;
                           });
                       }
                   })
                   .ConfigureLogging(builder =>
                   {
                         // optionally configure logging
                         builder.SetMinimumLevel(LogLevel.Trace);
-                  }).Build();
-
+                  }).Build();
+
+        private static X509Certificate2 GetCertificate()
+        {
+            var assm = Assembly.GetExecutingAssembly();
+            var binPath = System.IO.Path.GetDirectoryName(assm.GetName().CodeBase);
+            string certName = "FactoryServer.pfx";
+            string certPath = binPath.Substring(6) + "\\" + certName;
+            X509Certificate2 sslCert;

```

Comment:
this path checking wont work on linux, you would need to use Path.DirectorySeparatorChar. that said...

When we talked about implementing this before, the linux PR wasnt complete, with that PR there is a new way to configure service settings, which I think would be ideal to use here for consistency.

In [LoadOEMCustomizations()](https://github.com/microsoft/FactoryOrchestrator/blob/e65f63e46405d30bf86dbeb6d43c5c6f6c2db22d/src/Service/ServiceExe.cs#L2748) you will see that we now look for an appsettings.json. This file is checked for in a few places, including right next to the service executable. [If it exists, settings are parsed from it](https://www.tektutorialshub.com/asp-net-core/application-settings-appsettings-json-in-asp-net-core/).

Can we add a new setting named "SSLCertificateFile" or a better name you think of, then check for a cert there if found? You can pass it in [via CreateHost()](https://github.com/microsoft/FactoryOrchestrator/blob/e65f63e46405d30bf86dbeb6d43c5c6f6c2db22d/src/Service/ServiceExe.cs#L69) like we do for [IP & port](https://github.com/microsoft/FactoryOrchestrator/blob/e65f63e46405d30bf86dbeb6d43c5c6f6c2db22d/src/Service/ServiceExe.cs#L1619).

Reply:
happy to discuss on a call if you want to also